OCPBUGS-36495: Add runbook for ClusterMonitoringOperatorDeprecatedConfig

slashpai opened the pull request.

slashpai commented:
/jira refresh

machine424 commented on alerts/cluster-monitoring-operator/ClusterMonitoringOperatorDeprecatedConfig.md:
```diff
@@ -0,0 +1,33 @@
+
+Using a deprecated config is not recommended because the config has no effect and
+it may cause `cluster-monitoring-operator` go in degraded state when the config
+is removed in a future release.
```
```suggestion
is no longer supported in a future release.
```

machine424 commented on alerts/cluster-monitoring-operator/ClusterMonitoringOperatorDeprecatedConfig.md:
```diff
@@ -0,0 +1,33 @@
+  `Cluster Monitoring Operator is being used with deprecated configuration.`
+
+## Mitigation
+
```
I think it's better if we handle each situation individually:
- For the `k8sPrometheusAdapter.dedicatedServiceMonitors` field, you can remove the block, see the release notes (entry) at XXX (for clusters <4.16)
- For the `k8sPrometheusAdapter` field, check the release notes entry at YYY, some of the subfields may need to be migrated under `metricsServer` (or sth like this)

machine424 commented on alerts/cluster-monitoring-operator/ClusterMonitoringOperatorDeprecatedConfig.md:
```diff
@@ -0,0 +1,33 @@
+
+  `The configuration field k8sPrometheusAdapter in
+  openshift-monitoring/cluster-monitoring-config was deprecated in 4.16 and has
+  no effect, you can remove it.`
```
I think we should remove `you can remove it` from the alert description.
Now we have a runbook + in some cases we also need to migrate the config...

simonpasquier replied:
Ideally the user shouldn't even have to read the runbook so I'd update the alert description to say the same about migrating the config.

machine424 reviewed (commented):
Thanks JP!

juzhao commented:
do we need to add the runbook_url to ClusterMonitoringOperatorDeprecatedConfig alert?

slashpai commented:
> do we need to add the runbook_url to ClusterMonitoringOperatorDeprecatedConfig alert?

We will add once this is merged

simonpasquier commented on alerts/cluster-monitoring-operator/ClusterMonitoringOperatorDeprecatedConfig.md:
```diff
@@ -0,0 +1,33 @@
+## Impact
+
+Using a deprecated config is not recommended because the config has no effect and
+it may cause `cluster-monitoring-operator` go in degraded state when the config
```
Before we remove the field, we'll block the upgrades using the Upgradeable condition.

```suggestion
it may cause the Cluster Monitoring operator `Upgradeable` condition to become `False`  in a future OCP release.
```

simonpasquier commented on alerts/cluster-monitoring-operator/ClusterMonitoringOperatorDeprecatedConfig.md:
```diff
@@ -0,0 +1,33 @@
+
+## Diagnosis
+
+- Check the __Description__ and __Summary__ sections of the alert to identify
```
```suggestion
- Check the __Description__ and __Summary__ annotations of the alert to identify
```

simonpasquier commented:
cc @ssonigra @coffeegoesincodecomesout

slashpai commented:
addressed review comments.
@machine424 @simonpasquier  can you review again?
I will create PR to cmo to update alert description along with runbook annotation

slashpai commented:
@jan--f @machine424 could you please have another look on this?
